### PR TITLE
CDAP-7718 remove StreamBatchReadable

### PR DESCRIPTION
**Pull request by albertshau**
Removing StreamBatchReadable, which has been deprecated since 3.4

### CODE REVIEW

**Comment by albertshau** [comment]
http://builds.cask.co/browse/CDAP-DUT5141

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/ApplicationRegistrationStage.java [hunk 1]
```diff
@@ -84,14 +80,7 @@ private void registerDatasets(ApplicationWithPrograms input) {
...
       for (String dataset : program.getDataSets()) {
-        if (!dataset.startsWith(Constants.Stream.URL_PREFIX)) {
-          usageRegistry.register(programId, namespaceId.dataset(dataset));
-        }
-      }
-      String inputDatasetName = program.getInputDataSet();
-      if (inputDatasetName != null && inputDatasetName.startsWith(Constants.Stream.URL_PREFIX)) {
-        StreamBatchReadable stream = new StreamBatchReadable(URI.create(inputDatasetName));
-        usageRegistry.register(programId, namespaceId.stream(stream.getStreamName()));
+        usageRegistry.register(programId, namespaceId.dataset(dataset));
```
Comment (anew) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/ApplicationRegistrationStage.java:83]: does this mean that stream usage is not registered any more?

**Review by anew — COMMENTED** [review]
Looks good except one comments and this:
- Does StreamBatchReadable appear anywhere in the docs? guides? apps?

File: cdap-api/src/main/java/co/cask/cdap/api/data/stream/StreamBatchReadable.java [hunk 2]
```diff
@@ -1,339 +0,0 @@
...
- *       public void beforeSubmit(MapReduceContext context) {
- *         context.addInput(Input.ofStream("mystream"));
- *       }
- *   }
- * </code></pre>
- *
- * @deprecated as of version 3.4.0. Use {@link MapReduceContext#addInput(Input)}
- */
-@Deprecated
-public class StreamBatchReadable implements BatchReadable<Long, String> {
```
Comment (anwar6953) [inline cdap-api/src/main/java/co/cask/cdap/api/data/stream/StreamBatchReadable.java:61]: I thought there was a lot of logic in StreamBatchReadable that we were still using internally.
Is most of that logic not even needed?!?

Comment (albertshau) [reply cdap-api/src/main/java/co/cask/cdap/api/data/stream/StreamBatchReadable.java:61]: most of it centers around converting the required information into a URI, which was only needed back when you could set a stream as input in the configure method of a mapreduce. Today its only possible at initialize time so none of it is needed anymore.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/ApplicationRegistrationStage.java (hunk 1 above)
Comment (albertshau) [reply cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/ApplicationRegistrationStage.java:83]: yeah, it doesn't work today either. I opened CDAP-7720 about this

File: cdap-api/src/main/java/co/cask/cdap/api/data/stream/StreamBatchReadable.java (hunk 2 above)
Comment (anwar6953) [reply cdap-api/src/main/java/co/cask/cdap/api/data/stream/StreamBatchReadable.java:61]: gotcha. Cool.

**Comment by albertshau** [comment]
not in docs, not in apps, but found it in guides. Will fix those before merging this.

**Comment by albertshau** [comment]
fixed the workflow guide which was using it. build passed, merging.

**Pull request merged**